### ISSUE TITLE
Domain Isolated Web Part fails to load in SingleWebPartAppPage

### ISSUE THREAD

**Issue by PaoloPia** [SharePoint/sp-dev-docs#5357]
## Category

- [x] Bug

## Describe the bug

A domain isolated web part does not render in a modern page configured as _SingleWebPartAppPage_. Instead, you get a nasty exception with the following error message:

**Error: ***The value for "IFramedWebPartController is undefined" must not be undefined
in t in t in o in t in t in t in t in t in t in section in div in div in t in Styledt in t in o in t in t in t**

## Steps to reproduce

1. Create a new SPFx project (I'm using SPFx 1.10.0) with a Web Part and configure it to be domain isolated ("isDomainIsolated": true)
2. Bundle and package the solution, and install the SPPKG into the AppCatalog
3. Add the Web Part to a modern page
4. Configure the page to have the SingleWebPartAppPage layout (https://docs.microsoft.com/en-us/sharepoint/dev/spfx/web-parts/single-part-app-pages#changing-page-layout-using-pnp-powershell)
5. Reload the page and it will fail with the above error message

If I revert the page back to layout _Article_, the web part is back and fully functional.

## Expected behavior

I should be able to see the web part, instead of the error message.

## Developer environment

- OS: Windows 10
- Framework: SPFx 1.10.0
- Browser(s): Chrome Version 80.0.3987.106 (Official Build) (64-bit)
- Tooling: VSCode

Thanks!

**Comment by giovanibm** [comment]
i'm with the same error here, this is how i get the error:
- publish an app with the option 'Tenant Deployed' unchecked (in my case i'm publish through ci/cd in Azure Dev Ops)
- add the app to a site
- add a app through the cog -> add page -> tab app

If i only add the app to a page, like any other app, it works fine!

**Comment by joselu1sc** [comment]
Hello,

The same error when I am updating a normal webpart to be a SharePoint full page in SharePoint Online.

**Comment by ABoedt** [comment]
Having the same problem

**Comment by TheHailender** [comment]
Unfortunately I have the same problem with a React SharePointFullPage webpart. After a reload the page works.

**Comment by michelcarlo** [comment]
I'm having the same issue. Webpart used to work fine as a _SingleWebPartAppPage_, since some weeks ago it stopped working, and I'm getting the same error.
Still works as normal WebPart inserted in a page though.

**Comment by holmesrm23** [comment]
I'm experiencing this bug as well. Is it just forgotten about?

**Comment by cwdata** [comment]
Same problem here. Reloading the page does NOT help.

**Comment by JeroenGMI** [comment]
Same issue here. Currently hosting the webparts on a different page (not maximized).

**Comment by holmesrm23** [comment]
Is there any sort of work around? Some way to make the webpart pane wider in a non-SPA page even?

**Comment by ozippy** [comment]
I have the same issue. Created a web part with the generator, added the options to the manifest and deployed to SharePoint Online.
"supportedHosts": ["SharePointWebPart", "SharePointFullPage", "TeamsTab"],

**Comment by iiunknown** [comment]
Me too, have there any update about the issue？

**Comment by ChrisRomp** [comment]
Is one possible reason this isn't getting any attention is that it's in the sp-dev-docs (documentation) repo for some reason?

**Comment by rohanj1132** [comment]
`Will the components in the solution require permissions to access web APIs that are unique and not shared with other components in the tenant? (y/N)`
If you choose No to the above option while generating the project using yeoman scaffolding, I do not get the error.

However I am quite new to MS ecosystem and do not understand the implications of above choice. Either way this bug should be looked at.

**Comment by sunnyEspireGit** [comment]
@PaoloPia : i have found a solution for the same, basically we change page layout to "SingleWebPartAppPage" to add some webpart which should display on full page and gives us a cleaner look, so follow below steps ::

        1. Create a page
	2. Edit page 
	3. Add any custom/OOTB webpart 
	4. publish it
	5. and then change the layout of page 

that way you wont get that error.

**Comment by sunnyEspireGit** [comment]
Also you can use below powershell script to do all at once ::

$siteURL = "<siteUrl>"  

    # If you do not wish to pass credentials hard coded then you can use: -Credentials (Get-Credential). This will prompt to enter credentials  
    Connect-PnPOnline -Url $siteURL -UseWebLogin 
 
#Create Site Page
    $page_name = Add-PnPClientSidePage -Name "<pagename>"
 
#Add a new sections to the page
    Add-PnPClientSidePageSection -Page $page_name -SectionTemplate OneColumnVerticalSection -Order 1
 
#Add Hero webpart to page
    Add-PnPClientSideWebPart -Page $page_name -Component "<webpartName>" -Section 1 -Column 1
 
  Set-PnPClientSidePage -Identity $page_name -LayoutType "SingleWebPartAppPage"



Happy coding :)

**Comment by ahmed-asiri** [comment]
@AJIXuMuK Am facing the same issue today, I don't think its fixed yet,

Isolated web-parts does not work on the following:

1. Chrome Incognito
2. Safari
3. Firefox

I hope it will be fixed soon, cause it's really important for securing API calls
Thanks

**Comment by AJIXuMuK** [comment]
Hi @ahmed-asiri,
I believe, it's a completely separate issue.
In incognito mode iframe communications are blocked by default. If you enable cookies in Incognito - it will work,

**Comment by ahmed-asiri** [comment]
Thanks @AJIXuMuK  for your response,
your are right and That make sense, 
what about Safari and Firefox, is it the same issue?

**Comment by AJIXuMuK** [comment]
@ahmed-asiri - it is.
For Safari - it is default behavior. For FF - I assume you have a custom configuration to disable 3p cookies (by default, FF doesn't block them and Isolated Web Parts work with no issues).

**Comment by ahmed-asiri** [comment]
Thank you so much @AJIXuMuK,

Much Appreciated.

**Comment by patmill** [comment]
Closing this specific issue, as the issue (domain isolated webparts don't load) is solved.  There is a separate issue around incognito / 3p cookie blocking scenarios that we are working on.
